### PR TITLE
Improve/api

### PR DESCRIPTION
**Pull request by zenangst**
- [x] Expose collapseFieldsInSection:collectionView:
- [x] Add sectionIsCollapsed
- [x] Add support for custom headers
- [x] Add tests

### CODE REVIEW

**Comment by 3lvis** [comment]
Is this WIP?

**Comment by zenangst** [comment]
@3lvis yah, sorry. Fixed now

File: Source/FORMDataSource.m [hunk 1]
```diff
@@ -752,6 +756,15 @@ - (void)insertedIndexPathsAndSectionIndexForSection:(FORMSection *)section
     }
 }
 
+- (BOOL)sectionIsCollapsed:(NSInteger)section
```
Comment (kostiakoval) [inline Source/FORMDataSource.m]: this should work `return [self.collapsedGroups containsObject:@(section)]`

File: Source/FORMDataSource.h [hunk 2]
```diff
@@ -207,6 +207,12 @@ includingHiddenFields:(BOOL)includingHiddenFields
                                 NSArray *indexPaths))completion;
 
 /*!
+ * @return @c YES if the Section is collapsed.
+ * @param sectionID The section index for the section.
+ */
+- (BOOL)sectionIsCollapsed:(NSInteger)section;
```
Comment (3lvis) [inline Source/FORMDataSource.h]: I don't think this is the right naming.

This should be

```objc
- (BOOL)groupIsCollapsed:(NSInteger)groupID;
```

File: Source/FORMDataSource.h [hunk 3]
```diff
@@ -217,4 +223,11 @@ includingHiddenFields:(BOOL)includingHiddenFields
  */
 - (void)reloadFieldsAtIndexPaths:(NSArray *)indexPaths;
 
+/*!
+ * @param section A section index for the current section
+ * @param collectionView The used collectionView, usually from a @c UICollectionViewController.
+ */
+- (void)collapseFieldsInSection:(NSInteger)section
```
Comment (3lvis) [inline Source/FORMDataSource.h]: I don't think this is the right naming.

This should be

```objc
- (void)collapseGroup:(NSInteger)groupID; (we don't need the collectionView)
```

**Comment by 3lvis** [comment]
@zenangst Would you mind adding a few tests for this?

File: Source/FORMDataSource.h (hunk 3 above)
Comment (zenangst) [inline Source/FORMDataSource.h]: I like it, but it's not really the group ID, it's the group index.

Comment (3lvis) [inline Source/FORMDataSource.h]: Might be better if we make it use the group ID instead of the index.

**Comment by zenangst** [comment]
@3lvis of course! :)

File: Source/FORMDataSource.h (hunk 3 above)
Comment (3lvis) [inline Source/FORMDataSource.h]: @zenangst Read comments above

Comment (3lvis) [inline Source/FORMDataSource.h]: @zenangst Doesn't seem that you need this method to be public.

File: Source/FORMDataSource.h [hunk 4]
```diff
@@ -207,6 +218,12 @@ includingHiddenFields:(BOOL)includingHiddenFields
                                 NSArray *indexPaths))completion;
 
 /*!
+ * @return @c YES if the Group is collapsed.
+ * @param group The group index for the group.
+ */
+- (BOOL)groupIsCollapsed:(NSInteger)group;
```
Comment (3lvis) [inline Source/FORMDataSource.h]: @zenangst Doesn't seem that you need this method to be public.

**Comment by 3lvis** [comment]
Fantastic job as always

**Pull request merged**